cmake/Arduino-Toolchain.cmake

```diff
@@ -67,6 +67,7 @@ else ()
     # Set default path if none is set
     find_arduino_sdk(arduino_sdk_path)
     set(ARDUINO_SDK_PATH "${arduino_sdk_path}" CACHE PATH "Arduino SDK Path")
+    set(ENV{ARDUINO_SDK_PATH} "${ARDUINO_SDK_PATH}")
```

Comment:
Hmm I would wrap it in a conditional that applies only in the case where the env var _isn't_ defined, as it's user's responsibility to set it.

Reply:
Agree. This behavior is really weird even the variable has been cached in the file CMakeCache.txt. I've checked this set won't affect the environment variables. I'll track on this to make a better solution.